stocks2.lua

```diff
@@ -0,0 +1,544 @@
+                }),
+                widgets.Label({
+                    text = "to be filled in...",
+                    frame = {t = 1, l = 0},
+                    view_id = "filters_label",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Shift-C", pen = COLOR_LIGHTGREEN}, {text = ": Clear All", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-E", pen = COLOR_LIGHTGREEN}, {text = ": Enable All", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-G", pen = COLOR_LIGHTGREEN}, {text = ": Toggle Grouping", pen = COLOR_WHITE}
+                    },
+                    frame = {t = 8, l = 0},
+                    view_id = "filters_label2",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "-+", pen = COLOR_LIGHTGREEN}, {text = ": Min Qual: ", pen = COLOR_WHITE}, {text = "tbd", pen = COLOR_BROWN}, NEWLINE,
+                        {text = "/*", pen = COLOR_LIGHTGREEN}, {text = ": Max Qual: ", pen = COLOR_WHITE}, {text = "tbd", pen = COLOR_BROWN}, NEWLINE,
+                        {text = "Shift-W", pen = COLOR_LIGHTGREEN}, {text = ": Min Wear: ", pen = COLOR_WHITE}, {text = "tbd", pen = COLOR_BROWN}
+                    },
+                    frame = {t = 12, l = 0},
+                    view_id = "filters_qual_wear",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Actions (", pen = COLOR_BROWN}, {text = "...", pen = COLOR_LIGHTGREEN}, {text = " Items)", pen = COLOR_BROWN}
+                    },
+                    frame = {t = 16, l = 0},
+                    view_id = "actions_item_count_label",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Shift-Z", pen = COLOR_LIGHTGREEN}, {text = ": Zoom    ", pen = COLOR_WHITE},
+                        {text = "-D", pen = COLOR_LIGHTGREEN}, {text = ": Dump", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-F", pen = COLOR_LIGHTGREEN}, {text = ": Forbid  ", pen = COLOR_WHITE},
+                        {text = "-M", pen = COLOR_LIGHTGREEN}, {text = ": Melt", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-T", pen = COLOR_LIGHTGREEN}, {text = ": Mark for Trade", pen = COLOR_GREY},
+                    },
+                    frame = {t = 17, l = 0},
+                    view_id = "actions_label",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Shift-A", pen = COLOR_LIGHTGREEN}, {text = ": Apply to: ", pen = COLOR_WHITE},
+                        {text = apply_to_labels[apply_to_mode + 1], pen = COLOR_BROWN},
+                    },
+                    frame = {t = 20, l = 0},
+                    view_id = "apply_to_label",
+                }),
```

Comment:
This "apply to" selector seems anachronistic (which it is -- the original stocks screen was not mouse-aware). Let me finish this initial review, but it seems to me we need to reconsider the design for both keyboard and mouse accessibility. I encourage you to play with the trade and pasture assignment screens to see how the most recent screens utilize the mouse and enable multi-select of items. The critical difference between this screen and those is that in those screens, there is only one action -- mark/unmark for trade or assign/unassign to pasture/restraint/cage/pit. I suspect here we'll need to have a two-stage process: select item(s), then perform action. You can indicate that an item is selected the same way that the trade screen does it: https://github.com/DFHack/scripts/blob/master/internal/caravan/movegoods.lua#L331